server.js

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env node
+var path = require('path'),
+	http = require('http'),
+	express = require('express'),
+	bodyParser = require('body-parser'),
+	config = require(path.join(__dirname, 'config.json')),
+	compression = require('compression'),
+	app = express(),
+	request = require('request'),
+	async = require('async'),
+	_teams = [],
+	fs = require('fs'),
+	glob = require('glob');
+
+
+app.set('port', process.env.VCAP_APP_PORT || 80);
+app.use(compression());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.get("/teams", getTeamInfo);
+app.get("/imgs/:year?", getImageNames);
+
+var server = http.createServer(app);
+
+function getImageNames(req, res) {
+	var year = req.param('year', null);
+	var img_path = path.join("web", "images");
+	
+	if(year) {
+		img_path = path.join(img_path, year);
+	}
+
+	img_path = path.join(img_path, "**", "*.jpg"); //for globbing
+
+	glob(img_path, function(err, names) {
+			console.log("Files: " + JSON.stringify(names));
+		res.send(names);
+	});
+}
+
+function getTeamInfo(req, res) {
+	var processed = [];
+
+	_teams.forEach(function(team) {
+		var teamObj = {
+			name: team.name,
+			members: team.attendee_count,
+			description: team.description,
+			captain: team.captain
+		};
+		processed.push(teamObj);
+	});
+
+	res.send(processed);
+}
+
+function removeDuplicates(list) {
+	var uniqueArr = list.filter(function(elem, pos) {
+		return list.indexOf(elem) == pos;
+	});
+	return uniqueArr;
+}
+
+function removeHiddenTeams(teams, attendees) {
+	var visible_teams = [],
+		answered = false,
+		approved = false,
+		description = null,
+		captain = null;
+
+	teams.forEach(function(team, i_team) {
+		attendees.forEach(function(user, i) {
+			answered = false;
+			approved = false;
+			description = null;
+			captain = null;
+			
+			if(team.creator.emails[0].email === user.profile.email) {
+				captain = {
+					name: user.profile.name,
+					email: user.profile.email
+				};
+
+				user.answers.forEach(function(question) {
```

Comment:
You've got like 3 nested loops here. Pulling these out into functions so there's not so much nesting would increase readability.